### PR TITLE
RATIS-1855 Fix some sonar code smell and bugs in ratis-server

### PR DESCRIPTION
**Pull request by OneSizeFitsQuorum**
see [jira1855](https://issues.apache.org/jira/browse/RATIS-1855)

### CODE REVIEW

**Comment by OneSizeFitsQuorum** [comment]
PTAL @szetszwo

File: ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageImpl.java [hunk 1]
```diff
@@ -39,7 +40,7 @@ public class RaftStorageImpl implements RaftStorage {
   private final StartupOption startupOption;
   private final CorruptionPolicy logCorruptionPolicy;
   private volatile StorageState state = StorageState.UNINITIALIZED;
-  private volatile RaftStorageMetadataFileImpl metaFile;
+  private final AtomicReference<RaftStorageMetadataFileImpl> metaFile = new AtomicReference<>();
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageImpl.java:43]: Let's create a class:
```java
  static class MetaFile {
    private final AtomicReference<RaftStorageMetadataFileImpl> ref = new AtomicReference<>();

    RaftStorageMetadataFile get() {
      return ref.get();
    }

    RaftStorageMetadataFile set(File file) {
      final RaftStorageMetadataFileImpl impl = new RaftStorageMetadataFileImpl(file);
      ref.set(impl);
      return impl;
    }
  }
```

File: ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageImpl.java [hunk 2]
```diff
@@ -91,9 +92,9 @@ public CorruptionPolicy getLogCorruptionPolicy() {
 
   private void format() throws IOException {
     storageDir.clearDirectory();
-    metaFile = new RaftStorageMetadataFileImpl(storageDir.getMetaFile());
-    metaFile.persist(RaftStorageMetadata.getDefault());
-    LOG.info("Storage directory " + storageDir.getRoot() + " has been successfully formatted.");
+    metaFile.set(new RaftStorageMetadataFileImpl(storageDir.getMetaFile()));
+    metaFile.get().persist(RaftStorageMetadata.getDefault());
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageImpl.java:96]: After created a class, this becomes a single llne
```java
    metaFile.set(storageDir.getMetaFile()).persist(RaftStorageMetadata.getDefault());
```

File: ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageImpl.java [hunk 3]
```diff
@@ -112,8 +113,8 @@ private StorageState analyzeAndRecoverStorage(boolean toLock) throws IOException
       if (!f.exists()) {
         throw new FileNotFoundException("Metadata file " + f + " does not exists.");
       }
-      metaFile = new RaftStorageMetadataFileImpl(f);
-      final RaftStorageMetadata metadata = metaFile.getMetadata();
+      metaFile.set(new RaftStorageMetadataFileImpl(f));
+      final RaftStorageMetadata metadata = metaFile.get().getMetadata();
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageImpl.java:117]: Similarly, this becomes a single llne
```java
      final RaftStorageMetadata metadata = metaFile.set(f).getMetadata();
```

File: ratis-server/src/main/java/org/apache/ratis/server/storage/StorageImplUtils.java [hunk 4]
```diff
@@ -150,7 +152,7 @@ private RaftStorageImpl format() throws IOException {
       }
       throw new IOException("Failed to FORMAT a new storage dir for " + storageDirName + " from " + dirsInConf);
     }
-
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/storage/StorageImplUtils.java:153]: Please keep this empty line.

File: ratis-server/src/main/java/org/apache/ratis/server/storage/StorageImplUtils.java [hunk 5]
```diff
@@ -130,13 +131,14 @@ RaftStorageImpl run() throws IOException {
       }
     }
 
+    @SuppressWarnings("java:S1181")
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/storage/StorageImplUtils.java:134]: Could you add a comment at the end for this and the other SuppressWarnings?
```java
@SuppressWarnings("java:S1181") // catch Throwable
```
```java
@SuppressWarnings("java:S2095") // return Closable
```

**Review by szetszwo — COMMENTED** [review]
@OneSizeFitsQuorum , thanks a lot for working on this!  Please see the comments inlined.

File: ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageImpl.java (hunk 1 above)
Comment (OneSizeFitsQuorum) [reply ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageImpl.java:43]: Good advice~

File: ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageImpl.java (hunk 2 above)
Comment (OneSizeFitsQuorum) [reply ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageImpl.java:96]: OK

File: ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageImpl.java (hunk 3 above)
Comment (OneSizeFitsQuorum) [reply ratis-server/src/main/java/org/apache/ratis/server/storage/RaftStorageImpl.java:117]: OK

File: ratis-server/src/main/java/org/apache/ratis/server/storage/StorageImplUtils.java (hunk 5 above)
Comment (OneSizeFitsQuorum) [reply ratis-server/src/main/java/org/apache/ratis/server/storage/StorageImplUtils.java:134]: OK

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Pull request merged**